src/app/core/guards/validate-role/validate-role.guard.ts

```diff
@@ -0,0 +1,15 @@
+import { inject } from '@angular/core';
+import { CanActivateFn, Router, UrlTree } from '@angular/router';
+import { KeycloakService } from '../../services/keycloak/keycloak.service';
+
+export const validateRoleGuard: CanActivateFn = async (): Promise<
+  boolean | UrlTree
+> => {
+  const keycloakService = inject(KeycloakService);
+  const router = inject(Router);
+  if (keycloakService.getUserDetails()?.roles.includes('admin')) {
+    return true;
+  } else {
+    return router.createUrlTree(['/app/dashboard']);
```

Comment:
Why create `router.createUrlTree` instead of  `router.navigate` or `router.replace`?

Reply:
Solved! Changed it to ```router.navigate```